File: Dev Priya Gupta/Array/169. Majority Element.cpp
//Using sorting
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        return nums[nums.size()/2];
    }
};

Time Complexity : O(n log(n))
Space Complexity : O(1)

//Boyer-Moore Voting Algorithm
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int m=nums[0];
        int c=1;
        for(int i=1;i<nums.size();i++)
        {
            if(c==0)
            {
                m=nums[i];
                c=1;
            }
            else if(m==nums[i])
                c++;
            else
                c--;
        }
        return m;
    }
};

Time Complexity : O(n)
Space Complexity : O(1)

File: Vishakha Mishra /ARRAY/169. Majority Element.java
class Solution {
    public int majorityElement(int[] nums) {
        // Arrays.sort(nums);
        // int n = nums.length;
        // return nums[n/2];
        HashMap<Integer, Integer> map = new HashMap<>();
        int n = nums.length;
         for (int num : nums) {
            map.put(num, map.getOrDefault(num, 0) + 1);
        }
        for (int key : map.keySet()) {
            if (map.get(key) > n / 2) {
                return key;
            }
        }
        return -1;
    }
}
